MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountRepository.java

```diff
@@ -34,18 +40,35 @@
      * @return A new AccountRepository object.
      */
     public static AccountRepository create(AccountDao accountDao, Executor executor,
-                                           WebService webService) {
-        return new AccountRepository(accountDao, executor, webService);
+                                           WebService webService, Clock clock) {
+        return new AccountRepository(accountDao, executor, webService, clock);
     }
 
     /**
-     * Adds the given account name into the accounts table. If the name already exists,
-     * it is ignored. The account record is created with is_current set to false.
+     * Adds the given account name into the accounts table. If the name already exists, the account
+     * will be updated. The account is first sent to the server. If the server saves the account
+     * successfully, then the account will be saved locally with isPending set to false. Otherwise,
+     * it will be saved locally with isPending set to true.
      * @param name The name of the account to add.
      */
     void addAccount(String name) {
-        executor.execute(() -> {
-            accountDao.addAccount(name);
+        List<Account> accountList = Arrays.asList(Account.builder().setName(name)
```

Comment:
I think maybe this was a design decision earlier? It seems a bit confusion and unnecessary here that the repository only supports adding one account by name, but the request put it in a list, and the response handling for success and error calls different method in account dao?

Reply:
Yeah that's fair. I think I can make it less confusing by adding another endpoint on the server that takes in just an account name and timestamp

Reply:
Do you want to do that here, or in another pr?

Reply:
I'll do it here